Firestore/core/src/firebase/firestore/remote/grpc_completion.h

```diff
@@ -57,6 +57,8 @@ namespace remote {
  */
 class GrpcCompletion {
  public:
+  enum class Tag { Start, Read, Write, Finish };
```

Comment:
Comments? You actually wrote a nice one in the CL description--it would have been better added here :-).

Also, `Tag` is a somewhat confusing name given that gRPC itself has a concept of "tag" and we're passing pointers of instances of `GrpcCompletion` as gRPC tags.

Consider "kind" or "type" as alternative concepts.